### PR TITLE
Custom collection domain

### PR DESCRIPTION
**Pull request by bmulligan-kg**
Add ability to set custom collection domain for event traffic

### CODE REVIEW

**Review by StephanieZuniga — APPROVED** [review]

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 1]
```diff
@@ -176,6 +168,10 @@ + (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlaye
         MUXSDKAVPlayerViewControllerBinding *newBinding = [[MUXSDKAVPlayerViewControllerBinding alloc] initWithName:name software:MuxPlayerSoftwareAVPlayerViewController andView:player];
         [newBinding setAutomaticErrorTracking:automaticErrorTracking];
         newBinding.playDispatchDelegate = _playerBindingManager;
+        if (domain != nil && domain.length > 0) {
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:171]: Can we log an error message if the domain is invalid?

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 2]
```diff
@@ -308,6 +318,10 @@ + (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerLayer:(nonnull AVPlayerLayer *)
         MUXSDKAVPlayerLayerBinding *newBinding = [[MUXSDKAVPlayerLayerBinding alloc] initWithName:name software:MuxPlayerSoftwareAVPlayerLayer andView:player];
         newBinding.playDispatchDelegate = _playerBindingManager;
         [newBinding setAutomaticErrorTracking:automaticErrorTracking];
+        if (domain != nil && domain.length > 0) {
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:321]: same comment here

**Review by nidhik — APPROVED** [review]

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m (hunk 1 above)
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:171]: also, if they do not set a custom collection domain we want to continue to use litix.io - I might be missing it but where does that happen right now?

Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:171]: Oh, it happens in Core-  okay that makes sense https://github.com/muxinc/mux-stats-sdk-objc/pull/119/files#diff-1a15f01bbb046d01e3718e1e9506e210bd76412b3e8b85872e064e69266f9a82R54

Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:171]: regarding the "invalid" domain bit, we don't do any domain validation in either StatsSDK or CoreSDK (which matches the behaviour of the android & web SDKs). Do we want to add validation in CoreSDK? Steph brought this up too in the CoreSDK PR

Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:171]: the error is now being logged in CoreSDK

**Review by StephanieZuniga — APPROVED** [review]

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 3]
```diff
@@ -328,7 +342,28 @@ + (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerLayer:(nonnull AVPlayerLayer *)
...
+
++ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerLayer:(nonnull AVPlayerLayer *)player
+                                        withPlayerName:(nonnull NSString *)name
+                                          customerData:(nonnull MUXSDKCustomerData *)customerData
+                                automaticErrorTracking:(BOOL)automaticErrorTracking {
+
+    return [self monitorAVPlayerLayer:player
+                       withPlayerName:name
+                         customerData:customerData
+               automaticErrorTracking:true
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:355]: this should actually pass in the value of `automaticErrorTracking`

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 4]
```diff
@@ -198,6 +194,30 @@ + (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlaye
...
+
+
++ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlayerViewController *)player
+                                                 withPlayerName:(nonnull NSString *)name
+                                                   customerData:(nonnull MUXSDKCustomerData *)customerData
+                                         automaticErrorTracking:(BOOL)automaticErrorTracking {
+    return [self monitorAVPlayerViewController:player
+                                withPlayerName:name
+                                  customerData:customerData
+                        automaticErrorTracking:true
```
Comment (nidhik) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:206]: this should pass in the value of `automaticErrorTracking`

**Review by nidhik — APPROVED** [review]
lgtm pending those 2 updates to the monitor methods

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m (hunk 3 above)
Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:355]: good catch! i will update today

**Review by nidhik — APPROVED** [review]

**Pull request merged**